cases/caseworker_serializers.py

```
@@ -53,6 +61,39 @@ def validate_title(self, value):
             raise serializers.ValidationError("Title must be a non-empty string")
         return value
 
+    def validate_end_date(self, value):
+        try:
+            datetime.fromisoformat(value)
+        except (ValueError, TypeError):
+            raise serializers.ValidationError(
+                "Invalid end_date format (expected ISO format YYYY-MM-DD)"
+            )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)